airbrakes/imu/imu_data_packet.py

```diff
@@ -0,0 +1,152 @@
+"""Module for describing the datapackets from the IMU"""
+
+
+def _mro_slots(obj) -> list[str]:
+    """Helper function to get all __slots__ from the MRO (Method Resolution Order) of an object"""
+    return [
+        attr
+        for cls in obj.__class__.__mro__[:-1]  # :-1 doesn't include the object class
+        for attr in cls.__slots__
+    ]
+
+
+class IMUDataPacket:
+    """
+    Base class representing a collection of data packets from the IMU.
+    The attributes should be named the same as they are when sent from the IMU -- this just means
+    they're going to be in camelCase.
+
+    Args:
+        timestamp (int): The timestamp of the data packet in nanoseconds.
+    """
+
+    __slots__ = ("timestamp",)
+
+    def __init__(self, timestamp: int):
+        self.timestamp = timestamp
+
+    def __str__(self):
+        attributes = ", ".join(f"{attr}={getattr(self, attr)}" for attr in _mro_slots(self))
+        return f"{self.__class__.__name__}({attributes})"
+
+
+class RawDataPacket(IMUDataPacket):
+    """
+    Represents a raw data packet from the IMU. These values are exactly what the IMU read, without any processing.
+    It contains a timestamp and the raw values of the acceleration, gyroscope, and GPS correlation data.
+    """
+
+    __slots__ = (
+        "gpsCorrelTimestampFlags",
+        "gpsCorrelTimestampTow",  # Time of week
+        "gpsCorrelTimestampWeekNum",  # Week number
+        "scaledAccelX",
+        "scaledAccelY",
+        "scaledAccelZ",
+        "scaledGyroX",
+        "scaledGyroY",
+        "scaledGyroZ",
+    )
+
+    def __init__(
+        self,
+        timestamp: int,
+        gpsCorrelTimestampFlags: int | None = None,
+        gpsCorrelTimestampTow: float | None = None,
+        gpsCorrelTimestampWeekNum: int | None = None,
+        scaledAccelX: float | None = None,
+        scaledAccelY: float | None = None,
+        scaledAccelZ: float | None = None,
+        scaledGyroX: float | None = None,
+        scaledGyroY: float | None = None,
+        scaledGyroZ: float | None = None,
+    ):
+        super().__init__(timestamp)
+
+        self.gpsCorrelTimestampFlags = gpsCorrelTimestampFlags
+        self.gpsCorrelTimestampTow = gpsCorrelTimestampTow
+        self.gpsCorrelTimestampWeekNum = gpsCorrelTimestampWeekNum
+        self.scaledAccelX = scaledAccelX
+        self.scaledAccelY = scaledAccelY
+        self.scaledAccelZ = scaledAccelZ
+        self.scaledGyroX = scaledGyroX
+        self.scaledGyroY = scaledGyroY
+        self.scaledGyroZ = scaledGyroZ
+
+
+class EstimatedDataPacket(IMUDataPacket):
```

Comment:
Could maybe use `msgspec` Structs (for speed) or a `dataclass` with slots enabled (for readability)

Reply:
https://jcristharif.com/msgspec/structs.html

Reply:
yeah i'm fine with dataclass with slots for those classes. we aren't doing json (de)serialization so I don't see the need for that

Reply:
ok i'm down to use msgspec, in a different PR tho